matrix_client/client.py

```diff
@@ -84,23 +125,61 @@ def login_with_password(self, username, password, limit=1):
         return self.token
 
     def create_room(self, alias=None, is_public=False, invitees=()):
+        """ Create a new room on the homeserver.
+
+        Args:
+            alias (string): The canonical_alias of the room.
+            is_public (bool):  The public/private visibility of the room.
+            invitees (string[]): A set of user ids to invite into the room.
+
+        Returns:
+            Room
+
+        Raises:
+            MatrixRequestError
+        """
         response = self.api.create_room(alias, is_public, invitees)
         return self._mkroom(response["room_id"])
 
     def join_room(self, room_id_or_alias):
+        """ Join a room.
+
+        Args:
+            room_id_or_alias (string): Room ID or an alias.
+
+        Raises:
+            MatrixRequestError
+        """
         response = self.api.join_room(room_id_or_alias)
         room_id = (
             response["room_id"] if "room_id" in response else room_id_or_alias
         )
         return self._mkroom(room_id)
 
     def get_rooms(self):
+        """ Return a list of Room objects that the user has joined.
+
+        Returns:
+            Room[]: Rooms the user has joined.
+
+        """
         return self.rooms
 
     def add_listener(self, callback):
+        """ Add a listener that will send a callback when the client recieves
+            an event.
+        Args:
+            callback (func(roomchunk)): Callback called when an event arrives.
+        """
         self.listeners.append(callback)
 
     def listen_for_events(self, timeout=30000):
+        """ Listen once for events. Use listen_forever to block indefinitely.
+
+        Args:
+            timeout (int): How long to poll the Home Server for before
+                           retrying.
```

Comment:
Be consistent with leading indentation. Choose either a single indent or to match them up (I generally find the former easier)